teamscale-gradle-plugin/src/main/kotlin/com/teamscale/TestImpacted.kt

```diff
@@ -98,10 +101,19 @@ open class TestImpacted : Test() {
         }
     }
 
-    /** Specifies that the JUnit Platform should be used, but only impacted tests are executed. */
-    fun useImpactedJUnitPlatform(testFrameworkConfigure: Action<in JUnitPlatformOptions>) {
+    /** Overrides default behavior to only execute impacted tests. */
+    override fun useJUnitPlatform(testFrameworkConfigure: Action<in JUnitPlatformOptions>) {
         testFrameworkConfigure.execute(junitPlatformOptions)
 
+        if (includeEngines.isEmpty()) {
+            // Set the default engines as included engines
+            includeEngines = setOf("junit-jupiter", "junit-vintage")
+            junitPlatformOptions.includeEngines = includeEngines
+        }
+        if (junitPlatformOptions.excludeEngines.contains(IMPACTED_TEST_ENGINE)) {
+            throw GradleException("Engine '$IMPACTED_TEST_ENGINE' can't be excluded in '$TestImpacted' Gradle task")
```

Comment:
Why the `$` sign?

Reply:
https://kotlinlang.org/docs/reference/basic-syntax.html#using-string-templates

Reply:
🙈 Stupid me